adapter/src/ethereum.rs

```diff
@@ -113,29 +115,17 @@ impl Adapter for EthereumAdapter {
         }
     }
 
-    fn verify(&self, signer: &str, state_root: &str, sig: &str) -> AdapterResult<bool> {
-        let (decoded_adress, decoded_signature) = match (hex::decode(signer), hex::decode(sig)) {
-            (Ok(address), Ok(sig)) => (address, sig),
-            _ => {
-                return Err(AdapterError::Signature(
-                    "invalid signature or address".to_string(),
-                ))
-            }
-        };
-
-        let address = Address::from_slice(&decoded_adress);
+    fn verify(&self, signer: &ValidatorId, state_root: &str, sig: &str) -> AdapterResult<bool> {
```

Comment:
we take reference here as well, so yet another reason to return a reference from `whoami`
Also `get_auth` takes a reference.